src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx

```diff
@@ -512,28 +596,33 @@ export class CatalogPlotComponent extends React.Component<WidgetProps> {
         const widgetsStore = this.widgetStore;
         const catalogWidgetStore = this.catalogWidgetStore;
         catalogStore.updateCatalogProfiles(this.catalogFileId);
-        profileStore.setSelectedPointIndices([], false);
-        catalogWidgetStore.setShowSelectedData(false);
-        widgetsStore.initLinearFitting();
-        widgetsStore.initStatistic();
+        profileStore?.setSelectedPointIndices([], false);
+        catalogWidgetStore?.setShowSelectedData(false);
+        widgetsStore?.initLinearFitting();
+        widgetsStore?.initStatistic();
         this.updateStatistic();
     };
 
     // Single source selected
     private onSingleSourceClick = (event: Readonly<Plotly.PlotMouseEvent>) => {
         const selectionMode: DragMode[] = ["select", "lasso"];
-        const inDragmode = selectionMode.includes(this.widgetStore.dragmode);
+        const widgetStore = this.widgetStore;
+        if (!widgetStore) {
+            return;
+        }
+
+        const inDragmode = selectionMode.includes(widgetStore.dragmode);
         const profileStore = this.profileStore;
         const catalogWidgetStore = this.catalogWidgetStore;
-        if (event?.points?.length > 0 && inDragmode) {
+        if (event?.points?.length > 0 && inDragmode && profileStore && catalogWidgetStore) {
```

Comment:
makes more sense to move these checks up to line 610 as well

`if (!widgetStore || !profileStore || !catalogWidgetStore)`